pennylane/ops/qubit/parametric_ops.py

```diff
@@ -2258,16 +2298,16 @@ def simplify(self):
         """Simplifies the gate into RX or RY gates if possible."""
         wires = self.wires
 
-        if np.allclose(np.mod(self.data[1], 2 * np.pi), 0) and np.allclose(
-            np.mod(self.data[0] + self.data[1], 2 * np.pi), 0
-        ):
+        phi, delta = qml.math.mod(self.data, 2 * np.pi)
+
+        if np.allclose(delta, 0) and np.allclose(np.mod(phi + delta, 2 * np.pi), 0):
```

Comment:
```suggestion
        if qml.math.allclose(delta, 0) and qml.math.allclose( (phi + delta) % (2 * np.pi), 0):
```

Looks like some of these methods are ignoring differentiability.  I just copied the code over from elsewhere without paying much attention to it.

Reply:
Made this change everywhere.